crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -152,6 +152,13 @@ def transition(inst, name, state)
       if node.crowbar["state"] != state
         @logger.debug("Crowbar transition: state has changed so we need to do stuff for #{name} to #{state}")
 
+        # Do not allow change to shutdown state from crowbar_upgrade
+        # (we need to reboot the nodes for upgrade, but without changing the state)
+        if node.crowbar["state"] == "crowbar_upgrade" && state == "shutdown"
+          @logger.debug("Crowbar transition: current state is crowbar_upgrade; change to #{state} not allowed")
+          return [200, { :name => name } ]
```

Comment:
Style/HashSyntax: Use the new Ruby 1.9 hash syntax. (https://github.com/bbatsov/ruby-style-guide#hash-literals)<br>Style/SpaceInsideBrackets: Space inside square brackets detected. (https://github.com/bbatsov/ruby-style-guide#no-spaces-braces)